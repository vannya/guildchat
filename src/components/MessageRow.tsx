import React from 'react';

export interface IMessageRowProps {
	text: string;
	displayType: 'outgoing' | 'incoming';
}

export const MessageRow = ({
	text,
	displayType
}: IMessageRowProps): JSX.Element => {
	return <div className={`message-row--${displayType}`}>{text}</div>;
};
