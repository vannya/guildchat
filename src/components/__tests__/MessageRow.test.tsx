import { cleanup, render } from '@testing-library/react';
import React from 'react';
import { MessageRow, IMessageRowProps } from '../MessageRow';

describe('MessageRow', () => {
	const getProps = (): IMessageRowProps => {
		return {
			text: 'Message',
			displayType: 'outgoing'
		};
	};

	it('should match snapshot', () => {
		const props = getProps();
		const { container } = render(<MessageRow {...props} />);
		expect(container).toMatchSnapshot();
		cleanup();
	});

	it('should display incoming message properly', () => {
		const props = getProps();
		props.displayType = 'incoming';
		const { getByText } = render(<MessageRow {...props} />);
		const messageRow = getByText(props.text);
		expect(messageRow.classList).toContain('message-row--incoming');
		cleanup();
	});

	it('should display outgoing message properly', () => {
		const props = getProps();
		const { getByText } = render(<MessageRow {...props} />);
		const messageRow = getByText(props.text);
		expect(messageRow.classList).toContain('message-row--outgoing');
		cleanup();
	});
});
