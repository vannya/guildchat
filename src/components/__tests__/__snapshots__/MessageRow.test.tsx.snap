// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageRow should match snapshot 1`] = `
<div>
  <div
    class="message-row--outgoing"
  >
    Message
  </div>
</div>
`;
